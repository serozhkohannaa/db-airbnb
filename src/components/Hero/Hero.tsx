import React, { FC } from "react";
import './Hero.scss';

interface Props {
  heroText: string;
}

const Hero: FC<Props> = ({heroText}) => {
  return <section className='hero'>
	<div className="container">
	  <div className="hero-text">
		<h1 data-testid='hero-text' className='headline'>
		  {heroText}
		</h1>
	  </div>
	  <div className="hero-img">
		img
	  </div>
	</div>
  </section>
}

export default Hero;