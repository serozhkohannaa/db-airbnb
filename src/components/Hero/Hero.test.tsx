import React from 'react'
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import Hero from "./Hero";
import TestRenderer from 'react-test-renderer';

it('renders headline', () => {
  const section = document.createElement("section");
  ReactDOM.render(<Hero heroText='Airbnb listenings and reviews'/>, section)
})

it('renders correct text', () => {
  const { getByTestId } = render(<Hero heroText='simple text'/>);
  expect(getByTestId('hero-text')).toHaveTextContent('simple text')
})

it(('matches snapshot'), () => {
  const tree = TestRenderer.create(<Hero heroText='Airbnb listenings and reviews'/>).toJSON();
  expect(tree).toMatchSnapshot();
})