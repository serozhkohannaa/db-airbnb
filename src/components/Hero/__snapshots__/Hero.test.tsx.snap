// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="hero"
>
  <div
    className="container"
  >
    <div
      className="hero-text"
    >
      <h1
        className="headline"
        data-testid="hero-text"
      >
        Airbnb listenings and reviews
      </h1>
    </div>
    <div
      className="hero-img"
    >
      img
    </div>
  </div>
</section>
`;
